python/cudf/cudf/core/column/column.py

```diff
@@ -333,30 +332,20 @@ def from_arrow(cls, array: pa.Array) -> ColumnBase:
         elif isinstance(array.type, ArrowIntervalType):
             return cudf.core.column.IntervalColumn.from_arrow(array)
 
-        data = pa.table([array], [None])
+        if isinstance(array, pa.ChunkedArray):
+            array = array.combine_chunks()
```

Comment:
Do we always have to combine chunks? IIRC the existing implementation works without combining in most cases, and I don't think combining is free performance-wise so we should avoid it if we can. I could be wrong though, or misremembering an earlier state of the code.

Reply:
Ah right. Yeah this will make a copy on the CPU side.

I see now in libcudf side we only support returning tables (and not columns) from an arrow stream. I was hoping to avoid the dance of putting the chunked array in a pyarrow table but I think the dance is worth avoiding a CPU copy

Reply:
Thanks yeah I think this is the right call for now. We could generalize the libcudf APIs in the future if that helps.